### ISSUE TITLE
`typescript/explicit-member-accessibility` false positive in mixed TS & JS codebase

### ISSUE THREAD

**Issue by scottohara** [nzakas/eslint-plugin-typescript#102]
**What version of TypeScript are you using?**
2.6.2

**What version of typescript-eslint-parser are you using?**
9.0.0

**What version of eslint-plugin-typescript are you using?**
0.8.1

**What code were you trying to parse?**

```javascript
/* foo.js */
export default class Foo {
  constructor() { ... }
}
```

```javascript
/* bar.ts */
export default class Bar {
  public constructor() { ... }
}
```

```
eslint --rule 'typescript/explicit-member-accessibility: 2' '*.{ts,js}'
```

**What did you expect to happen?**
No errors/warnings

**What happened?**

```
Missing accessibility modifier on method definition constructor
```

In a codebase that includes both JS and TS files, enabling the `typescript/explicit-member-accessibility` rule triggers false positives on all classes in JS files, because the class members don't have accessibility modifiers (e.g. `public`, `private`, `protected`); despite the fact that such modifiers are invalid syntax in Javascript.

**Comment by scottohara** [comment]
Could this be addressed by simply checking that the source file name ends with `*.ts`, or would just checking the file extension be too naive?

e.g.
```javascript
function checkMethodAccessibilityModifier(methodDefinition) {
  if (!methodDefinition.accessibility && /\.ts$/test(context.getFilename()) {
    ...
  }
}
```

If so, I'd be happy to submit a PR to this effect.
